src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -457,48 +711,54 @@ private BoundDecisionDag MakeDecisionDag(SyntaxNode syntax, ImmutableArray<Parti
 
             if (trueBuilder != null)
             {
-                var pcd = trueBuilder.Count == c.Decisions.Length ? c : new PartialCaseDecision(c.Index, c.Syntax, trueBuilder.ToImmutableAndFree(), c.Bindings, c.WhereClause, c.CaseLabel);
+                var pcd = trueBuilder.Count == c.Decisions.Length ? c : new PartialCaseDecision(c.Index, c.Syntax, trueBuilder.ToImmutableAndFree(), c.Bindings, c.WhenClause, c.CaseLabel);
                 whenTrueBuilder.Add(pcd);
             }
 
             if (falseBuilder != null)
             {
-                var pcd = falseBuilder.Count == c.Decisions.Length ? c : new PartialCaseDecision(c.Index, c.Syntax, falseBuilder.ToImmutableAndFree(), c.Bindings, c.WhereClause, c.CaseLabel);
+                var pcd = falseBuilder.Count == c.Decisions.Length ? c : new PartialCaseDecision(c.Index, c.Syntax, falseBuilder.ToImmutableAndFree(), c.Bindings, c.WhenClause, c.CaseLabel);
```

Comment:
>falseBuilder.Count == c.Decisions.Length ? c : new PartialCaseDecision(c.Index, c.Syntax, falseBuilder.ToImmutableAndFree(), c.Bindings, c.WhenClause, c.CaseLabel) [](start = 26, length = 163)

Perhaps extract a local function `createPartialCaseDecision(PartialCaseDecision decision, ArrayBuilder<BoundDagDecision> builder)` for use here and above.